app/util/concurrency.py

```diff
@@ -0,0 +1,64 @@
+from threading import Thread
+from datetime import datetime
+
+DEFAULT_TIMEOUT = 10
+
+"""
+Class which extends Thread to expose a result from the target (input) function.
+"""
+class ResultThread(Thread):
+    def run(self):
+        self.result = self._target()
+    def get_result(self):
+        return self.result
+
+def get_now():
+    return datetime.now().timestamp()
+
+def call_parallel(funcs, timeout=None):
+    """
+    Takes in a list of functions, which are then run asynchronously, and returns a list of their results.
+    Doctest below.
+
+    >>> from time import sleep
+
+    >>> def sleep_func_gen(timeout):
+    ...     def sleep_func():
+    ...         sleep(timeout)
+    ...         return timeout
+    ...     return sleep_func
+
+    >>> calls = [sleep_func_gen(5), sleep_func_gen(1), sleep_func_gen(3), sleep_func_gen(5),]
+
+    >>> call_parallel([])
+    []
+
+    >>> call_parallel(calls)
+    [5, 1, 3, 5]
+
+    >>> call_parallel(calls, timeout=2)
+    [None, 1, None, None]
+    """
+    if timeout == None:
```

Comment:
In the declaration, you can just do `call_parallel(funcs, timeout=DEFAULT_TIMEOUT)`

The `None` thing is useful if there's an object you want to assign (b/c python keeps using the same instance of the object), but for an integer it's fine.